play-services-nearby-core/src/main/kotlin/org/microg/gms/nearby/exposurenotification/ExposureDatabase.kt

```diff
@@ -901,6 +904,53 @@ class ExposureDatabase private constructor(private val context: Context) : SQLit
             }
         }
 
+        fun export(context: Context) {
+            // FileProvider cannot directly access /databases, so we have to copy the database first
+            // In addition, we filter out only the Advertisements table to not export sensitive TEKs
+
+            // Create a new database which will store only the Advertisements table
+            val exportDir = File(context.getCacheDir(), "exposureDatabase")
+            exportDir.mkdir()
+            val exportFile = File(exportDir, "database.db")
+            if (exportFile.delete()) {
+                Log.d("EN-DB-Exporter", "Deleted old export database.")
+            }
+            val db = SQLiteDatabase.openOrCreateDatabase(exportFile, null)
+
+            // Attach the full database to the new empty db
+            val databaseFile = context.getDatabasePath(DB_NAME);
+            db.execSQL("ATTACH '$databaseFile' AS fulldb;")
+
+            // copy TABLE_ADVERTISEMENTS over
+            db.execSQL("CREATE TABLE $TABLE_ADVERTISEMENTS AS SELECT * FROM fulldb.$TABLE_ADVERTISEMENTS;")
+
+            // Detach original db, close new db
+            db.execSQL("DETACH DATABASE fulldb;")
+            db.close()
+
+            // Use the FileProvider to get a content URI for the new DB
+            val fileUri: Uri? = try {
+                FileProvider.getUriForFile(context,"org.microg.gms.nearby.exposurenotification.export", exportFile)
+            } catch (e: IllegalArgumentException) {
+                Log.e("EN-DB-Exporter", "The database file can't be shared: $exportFile $e")
+                null
+            }
+
+            // Open a sharesheet
+            if (fileUri != null) {
+                // Grant temporary read permission to the content URI
+                val sendIntent: Intent = Intent().apply {
+                    action = Intent.ACTION_SEND
+                    putExtra(Intent.EXTRA_STREAM, fileUri)
+                    addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
+                    type = "application/vnd.sqlite3"
```

Comment:
Can we use a more specific file type such that https://github.com/mh-/corona-warn-companion-android/pull/97 can specifically listen for that file type? Suggestion: `application/vnd.microg.exposure+sqlite3`

Reply:
@tlambertz have seen this? any opinion?

Reply:
I do not have a strong opinion on this one. Using the default sqlite3 type would allow a user to open the database with an sqlite3 browser directly.. though I am not aware of any app that only listens to this specific type. I'll change it.